packages/sdk/src/modules/did/index.ts

```diff
@@ -152,24 +162,44 @@ export class Did extends Base {
     return `did:peaq:${address}`;
   }
 
-  private _createVerificationMethod(address: Address) {
+  private _createVerificationMethod(verification: Verification, address: Address) {
     const id = uuidv4();
     const verificationMethod = new peaqDidProto.VerificationMethod();
 
+    // does this id change if it is Ed25519 vs Sr25519?
     verificationMethod.setId(id);
-    verificationMethod.setType(
-      peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020
-    );
+
+    if (verification.type !== peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020 &&
```

Comment:
Do we allow for any type of verification method or only the ones as defined in the 'Verification Types' enum in the proto doc. This lines of codes make sure the type passed is either ed/sr25519.